components/eam/src/physics/cam/gw/gw_common.F90

```diff
@@ -320,161 +320,60 @@ end subroutine momentum_energy_conservation
 
 !==========================================================================
 
-subroutine gw_drag_prof(ncol, ngwv, src_level, tend_level, do_taper, dt, &
-     lat,           t,    ti,  pmid, pint, dpm,   rdpm, &
-     piln, rhoi,    nm,   ni,  ubm,  ubi,  xv,    yv,   &
-     effgw,      c, kvtt, q,   dse,  tau,  utgw,  vtgw, &
-     ttgw, qtgw, taucd,   egwdffi,   gwut, dttdf, dttke)
-
-  !-----------------------------------------------------------------------
-  ! Solve for the drag profile from the multiple gravity wave drag
-  ! parameterization.
-  ! 1. scan up from the wave source to determine the stress profile
-  ! 2. scan down the stress profile to determine the tendencies
-  !     => apply bounds to the tendency
-  !          a. from wkb solution
-  !          b. from computational stability constraints
-  !     => adjust stress on interface below to reflect actual bounded
-  !        tendency
-  !-----------------------------------------------------------------------
-
-  use gw_diffusion, only: gw_ediff, gw_diff_tend
-  use vdiff_lu_solver, only: lu_decomp
+subroutine compute_stress_profiles_and_diffusivities(ncol, ngwv, src_level, ubi, c, rhoi, ni, kvtt, &
```

Comment:
should we prepend the subroutine names with something like "gwd_"?

Reply:
Yeah, that is probably a good idea. All the other functions in this file have a prefix `gw_`